### PR TITLE
Rewrite of the CSI driver docs.

### PR DESCRIPTION
**Pull request by jeffbearer**
This document has been updated for clarity and to better address users needs regarding the Azure Managed Lustre CSI Driver.

### CODE REVIEW

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 1]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
+The option to peer two different virtual networks has the advantage of separating the management of the various networks to different privileged roles. Peering can also provide additional flexibility as it can be made across Azure subscriptions or regions. Virtual Network Peering will require coordination between the two networks to avoid choosing conflicting IP network spaces.
 
-  For sizing details, including a formula for calculating IP requirements, see [Configure Azure CNI networking in Azure Kubernetes Service](/azure/aks/configure-azure-cni).
+![Diagram showing two VNets, one for Azure Managed Lustre and one for AKS, with a VNet peering arrow connecting them.](media/use-csi-driver-kubernetes/subnet-access-option-2.png)
 
-- If you host the AKS cluster within your Azure Managed Lustre subnet, make sure your Lustre file system subnet has enough IP addresses to support both the Azure Managed Lustre system and the AKS cluster.
+### Installing AKS into a Subnet on the Azure Managed Lustre Virtual Network
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:76]:
```suggestion
### Installing AKS into a subnet on the Azure Managed Lustre virtual network
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 2]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
 
-### Option 2: Use Azure CNI in AKS, and peer the virtual networks
+The option to install the AKS cluster into the Azure Managed Lustre Virtual Network with the *Bring Your Own Network* feature in AKS can be advantageous where you want scenarios where the network is managed singularly. An additional subnet sized to meet your AKS networking requirements will need to be created in the Azure Managed Lustre virtual network.
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:78]:
```suggestion
The option to install the AKS cluster into the Azure Managed Lustre virtual network with the *Bring Your Own Network* feature in AKS can be advantageous where you want scenarios where the network is managed singularly. An additional subnet sized to meet your AKS networking requirements will need to be created in the Azure Managed Lustre virtual network.
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 3]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
 
-  For more information, see [Kubenet (basic) networking](/azure/aks/concepts-network#kubenet-basic-networking) in [Network concepts for applications in Azure Kubernetes Service (AKS)](/azure/aks/concepts-network).
+1. [Check the installation](#check-the-installation) by optionally using an echo pod to confirm the driver is working.
 
-- To learn how to connect the two networks, see [Virtual network peering](/azure/virtual-network/virtual-network-peering-overview).
+The following sections describe each task in greater detail.
 
 ## Create an Azure Managed Lustre file system
 
 If you haven't already created your Azure Managed Lustre File System cluster, create the cluster now. For instructions, see [Create an Azure Managed Lustre file system in the Azure portal](create-file-system-portal.md). Currently, the driver can only be used with a pre-existing Azure Managed Lustre system.
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:104]:
```suggestion
If you haven't already created your Azure Managed Lustre file system cluster, create the cluster now. For instructions, see [Create an Azure Managed Lustre file system in the Azure portal](create-file-system-portal.md). Currently, the driver can only be used with an existing Azure Managed Lustre file system.
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 4]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
 
-## Connect a client to your Azure Managed Lustre file system
+## Create an AKS Kubernetes Cluster
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:106]:
```suggestion
## Create an AKS cluster
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 5]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
+
+If you haven't already created your AKS cluster, create a cluster deployment, see [Deploy an Azure Kubernetes Service (AKS) cluster](/azure/aks/learn/quick-kubernetes-deploy-portal).
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:108]:
```suggestion
If you haven't already created your AKS cluster, create a cluster deployment. See [Deploy an Azure Kubernetes Service (AKS) cluster](/azure/aks/learn/quick-kubernetes-deploy-portal).
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 6]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
+
+## Create virtual network peering
 
-Do the following steps to install a Lustre client, and mount the file system:
+^[NOTE]
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:112]:
```suggestion
> [!NOTE]
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 7]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
+^Skip this network peering step if you installed AKS into a subnet on the Azure Managed Lustre virtual network.
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:113]:
```suggestion
> Skip this network peering step if you installed AKS into a subnet on the Azure Managed Lustre virtual network.
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 8]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
 
-1. Review the steps for installing the client software and mounting the system in [Connect to clients](connect-clients.md).
+The AKS virtual network is created in a separate resource group from the AKS cluster's resource group. You can find the name of this resource group by going to your AKS cluster in the Azure Portal choosing the Properties blade and finding the Infrastructure resource group.  This resource group contains the virtual network that needs to be paired with the Azure Managed Lustre virtual network. It matches the pattern **MC\_&lt;aks-rg-name&rt;\_&lt;aks-cluster-name&rt;\_&lt;region&rt;**.
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:115]:
```suggestion
The AKS virtual network is created in a separate resource group from the AKS cluster's resource group. You can find the name of this resource group by going to your AKS cluster in the Azure Portal choosing the **Properties** blade and finding the **Infrastructure** resource group.  This resource group contains the virtual network that needs to be paired with the Azure Managed Lustre virtual network. It matches the pattern **MC\_&lt;aks-rg-name&rt;\_&lt;aks-cluster-name&rt;\_&lt;region&rt;**.
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 9]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
 
-1. Link to specific steps for the client software version you're installing. See [Install client software](connect-clients.md#install-client-software) for links to supported software versions. Follow the steps in the individual article for your preferred client software version to install the client software.
+Consult [Virtual Network Peering](/azure/virtual-network/virtual-network-peering-overview) to peer the AKS virtual network with your Azure Manages Lustre virtual network.
+
+^[TIP]
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:119]:
```suggestion
> [!TIP]
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 10]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
+^Due to the naming of the MC_ resource groups and virtual networks names of networks can be similar or the same across multiple AKS deployments. When setting up peering pay close attention that you are choosing the AKS networks that you intend to choose.
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:120]:
```suggestion
> Due to the naming of the MC_ resource groups and virtual networks, names of networks can be similar or the same across multiple AKS deployments. When setting up peering pay close attention that you are choosing the AKS networks that you intend to choose.
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 11]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
 
 ## Connect to the AKS cluster
 
 Connect to the Azure Kubernetes Service cluster by doing these steps:
 
-1. Sign in to [the Azure portal](https://portal.azure.com).
+1. Open a terminal session on with access to the Azure CLI tools and login to your Azure Account.
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:126]:
```suggestion
1. Open a terminal session with access to the Azure CLI tools, and login to your Azure account.
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 12]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
-1. On the **Overview** page for your AKS cluster, select the **Get started** tab.
+   ```azurecli
+   az login
+   ```
 
-   To see CLI commands populated with your AKS cluster values, select **Connect**.
+1. Sign in to [the Azure portal](https://portal.azure.com).
 
-1. Open a PowerShell session on the system where you install and administer the CSI driver.
+1. Find your AKS cluster and load the **Overview** blade, select the **Connect** button then copy the command for **Download cluster credentials**.
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:134]:
```suggestion
1. Find your AKS cluster. Select the **Overview** blade, then select the **Connect** button and copy the command for **Download cluster credentials**.
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 13]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
-The Azure Managed Lustre container support interface (CSI) driver for Kubernetes enables you to use an Azure Managed Lustre storage system with Kubernetes containers. The driver is compatible with [Azure Kubernetes Service (AKS)](/azure/aks/).
-
-Kubernetes can simplify the process to configure and deploy virtual client endpoints for your Azure Managed Lustre workload. It can automate setup tasks like these:
-
-- Create Virtual Machine Scale Sets used by AKS to run the pods.
-- Load the correct Lustre client software onto the VM instances.
-- Specify the Azure Managed Lustre mount point, and propagate that information to the client pods.
-
-The Azure Managed Lustre CSI driver can automate the client software and mount tasks. The driver provides a CSI controller plugin as a deployment with two replicas, by default, and a CSI node plugin as a DaemonSet. You can change the number of replicas.
+The Azure Managed Lustre Continer Support Interface (CSI) Driver for AKS enables you to access Azure Managed Lustre storage as persistant storage volumes from Kubernetes containers deployed in [Azure Kubernetes Service (AKS)](/azure/aks/).
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:22]:
```suggestion
The Azure Managed Lustre Continer Support Interface (CSI) driver for AKS enables you to access Azure Managed Lustre storage as persistant storage volumes from Kubernetes containers deployed in [Azure Kubernetes Service (AKS)](/azure/aks/).
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 14]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
 
 ## Compatible Kubernetes versions
 
-The Azure Managed Lustre CSI Driver is compatible with [Azure Kubernetes Service](/azure/aks/). Other Kubernetes installations aren't currently supported.
+The Azure Managed Lustre CSI Driver for AKS is compatible with [Azure Kubernetes Service (AKS)](/azure/aks/). Other Kubernetes installations are not currently supported.
```
Comment (pauljewellmsft) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:26]:
```suggestion
The Azure Managed Lustre CSI driver for AKS is compatible with [Azure Kubernetes Service (AKS)](/azure/aks/). Other Kubernetes installations are not currently supported.
```

**Review by pauljewellmsft — COMMENTED** [review]
@jeffbearer Thanks for making these changes! I added several suggestions to the PR, which you can accept and commit to incorporate with the PR. These include fixes for build warnings, along with changes for consistency and conformity to doc standards. Thanks!

**Comment by Jak-MS** [comment]
@jeffbearer - after you've completed the changes, tag @pauljewellmsft in a comment for a review and approval. (and take the PR out of draft mode by selecting the 'Ready for review' button at the end of the PR. 

@pauljewellmsft - when this is approved and ready, can you add `#sign-off` in a comment?

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

**Comment by jeffbearer** [comment]
@pauljewellmsft  I incorporated your changes.
@dabradley  I'd like your review on this prior to merge.

File: azure-managed-lustre/use-csi-driver-kubernetes.md (hunk 3 above)
Comment (jeffbearer) [reply azure-managed-lustre/use-csi-driver-kubernetes.md:102]: incorporated

File: azure-managed-lustre/use-csi-driver-kubernetes.md (hunk 2 above)
Comment (jeffbearer) [reply azure-managed-lustre/use-csi-driver-kubernetes.md]: incorporated

File: azure-managed-lustre/use-csi-driver-kubernetes.md (hunk 1 above)
Comment (jeffbearer) [reply azure-managed-lustre/use-csi-driver-kubernetes.md:74]: incorporated

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 15]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
+## Plan your AKS Deployment
 
-1. [Check the installation](#check-the-installation) by optionally using an echo pod to confirm the driver is working.
+There are several different options when deploying Azure Kubernetes Service that affect the operation between AKS and Azure Managed Lustre.
 
-The following sections describe each task in greater detail.
+### Determine the network type to use with AKS
 
-## Provide subnet access between AKS and Azure Managed Lustre
+There are two network types that are compatible with the Ubuntu Linux OS SKU, kubenet and the Azure Container Network Interface (CNI) Driver. Both options work with the Azure Managed Lustre CSI Driver for AKS but they have different requirements that need to be understood when setting up virtual networking and AKS. See [Networking concepts for applications in Azure Kubernetes Service (AKS)](/azure/aks/concepts-network) for more information on determining the proper selection.
```
[the remark this answers is not in the document]
Comment (jeffbearer) [reply azure-managed-lustre/use-csi-driver-kubernetes.md]: incorporated

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 16]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
-## Setup overview
+<!---
+See [Upgrading Azure Managed Lustre CSI driver](/azure/azure-managed-lustre/csi-driver-upgrading) for more info on upgrading the CSI driver.
+--->
 
-To enable  the Azure Managed Lustre CSI Driver for Kubernetes, perform these steps:
+## Prerequisites
 
-1. [Provide subnet access between AKS and your Azure Managed Lustre file system](#provide-subnet-access-between-aks-and-azure-managed-lustre).
+Before you deploy the Azure Managed Lustre CSI Driver for AKS, complete the following prerequisites:
```
[the remark this answers is not in the document]
Comment (jeffbearer) [reply azure-managed-lustre/use-csi-driver-kubernetes.md]: incorporated

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 17]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
+> The Azure Managed Lustre CSI driver only works with the Ubuntu Linux OS SKU for Node Pools of AKS currently.
 
-## Prerequisites
+## Compatible Lustre versions
 
-Before you deploy the Azure Managed Lustre CSI Driver for Kubernetes, complete the following prerequisite:
+The Azure Managed Lustre CSI Driver for AKS is compatible with [Azure Managed Lustre](/azure/azure-managed-lustre). Other Lustre installations are not currently supported.  
 
-- Create the AKS cluster, if you haven't already. For instructions, see [Deploy an Azure Kubernetes Service (AKS) cluster](/azure/aks/learn/quick-kubernetes-deploy-portal).
+The Azure Managed Lustre CSI Driver versions 0.1.10 and later are supported with the current version of the Azure Managed Lustre Service.
```
[the remark this answers is not in the document]
Comment (jeffbearer) [reply azure-managed-lustre/use-csi-driver-kubernetes.md]: incorporated

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 18]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
...
-| main branch        | `mcr.microsoft.com/oss/kubernetes-csi/azurelustre-csi:latest`  | 1.21 or later | 2.15.1 |
-| v0.1.12            | `mcr.microsoft.com/oss/kubernetes-csi/azurelustre-csi:v0.1.12` | 1.21 or later | 2.15.1 |
+> [!IMPORTANT]
+> The Azure Managed Lustre CSI driver only works with the Ubuntu Linux OS SKU for Node Pools of AKS currently.
 
-## Prerequisites
+## Compatible Lustre versions
 
-Before you deploy the Azure Managed Lustre CSI Driver for Kubernetes, complete the following prerequisite:
+The Azure Managed Lustre CSI Driver for AKS is compatible with [Azure Managed Lustre](/azure/azure-managed-lustre). Other Lustre installations are not currently supported.
```
[the remark this answers is not in the document]
Comment (jeffbearer) [reply azure-managed-lustre/use-csi-driver-kubernetes.md]: incorporated

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 19]
```diff
@@ -13,129 +13,145 @@ ms.reviewer: brianl
 
 ---
 
-# Use the Azure Managed Lustre CSI Driver for Kubernetes
+# Use the Azure Managed Lustre CSI Driver with Azure Kubernetes Service
 
-The article describes how to install and mount the Azure Managed Lustre container support interface (CSI) driver for Kubernetes to enable you to use an Azure Managed Lustre storage system with your Kubernetes containers in an Azure Kubernetes Service (AKS) deployment.
+This article describes how to plan, install, and use [Azure Managed Lustre](/azure/azure-managed-lustre) in [Azure Kubernetes Service (AKS)](/azure/aks/) with the [Azure Managed Lustre Kubernetes container support interface driver (Azure Managed Lustre CSI Driver)](https://github.com/kubernetes-sigs/azurelustre-csi-driver).
```
[the remark this answers is not in the document]
Comment (jeffbearer) [reply azure-managed-lustre/use-csi-driver-kubernetes.md]: incorporated

File: azure-managed-lustre/use-csi-driver-kubernetes.md (hunk 4 above)
Comment (jeffbearer) [reply azure-managed-lustre/use-csi-driver-kubernetes.md:104]: incorporated

File: azure-managed-lustre/use-csi-driver-kubernetes.md (hunk 8 above)
Comment (jeffbearer) [reply azure-managed-lustre/use-csi-driver-kubernetes.md]: incorporated

**Comment by jeffbearer** [comment]
@pauljewellmsft  i'm done with the incessant churning.  take a look I think it's ready for merge.

**Review by pauljewellmsft — APPROVED** [review]
@jeffbearer Thanks so much for making the changes! I'll sign off shortly.

**Comment by pauljewellmsft** [comment]
#sign-off

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 20]
```diff
@@ -13,129 +13,143 @@ ms.reviewer: brianl
...
-1. Link to specific steps for the client software version you're installing. See [Install client software](connect-clients.md#install-client-software) for links to supported software versions. Follow the steps in the individual article for your preferred client software version to install the client software.
+> [TIP]
+> Due to the naming of the MC_ resource groups and virtual networks, names of networks can be similar or the same across multiple AKS deployments. When setting up peering pay close attention that you are choosing the AKS networks that you intend to choose.
 
 ## Connect to the AKS cluster
 
 Connect to the Azure Kubernetes Service cluster by doing these steps:
 
-1. Sign in to [the Azure portal](https://portal.azure.com).
+1. Open a terminal session with access to the Azure CLI tools, and login to your Azure account.
```
Comment (Jak-MS) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:124]:
```suggestion
1. Open a terminal session with access to the Azure CLI tools and log in to your Azure account.
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 21]
```diff
@@ -13,129 +13,143 @@ ms.reviewer: brianl
...
-Do the following steps to install a Lustre client, and mount the file system:
+The AKS virtual network is created in a separate resource group from the AKS cluster's resource group. You can find the name of this resource group by going to your AKS cluster in the Azure Portal choosing the **Properties** blade and finding the **Infrastructure** resource group.  This resource group contains the virtual network that needs to be paired with the Azure Managed Lustre virtual network. It matches the pattern **MC\_&lt;aks-rg-name&rt;\_&lt;aks-cluster-name&rt;\_&lt;region&rt;**.
 
-1. Review the steps for installing the client software and mounting the system in [Connect to clients](connect-clients.md).
+Consult [Virtual Network Peering](/azure/virtual-network/virtual-network-peering-overview) to peer the AKS virtual network with your Azure Manages Lustre virtual network.
 
-1. Link to specific steps for the client software version you're installing. See [Install client software](connect-clients.md#install-client-software) for links to supported software versions. Follow the steps in the individual article for your preferred client software version to install the client software.
+> [TIP]
```
Comment (Jak-MS) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:117]:
```suggestion
> [!TIP]
```

File: azure-managed-lustre/use-csi-driver-kubernetes.md [hunk 22]
```diff
@@ -13,129 +13,143 @@ ms.reviewer: brianl
...
 
-## Connect a client to your Azure Managed Lustre file system
+## Create virtual network peering
 
-Do the following steps to install a Lustre client, and mount the file system:
+> [!NOTE]
+> Skip this network peering step if you installed AKS into a subnet on the Azure Managed Lustre virtual network.
 
-1. Review the steps for installing the client software and mounting the system in [Connect to clients](connect-clients.md).
+The AKS virtual network is created in a separate resource group from the AKS cluster's resource group. You can find the name of this resource group by going to your AKS cluster in the Azure Portal choosing the **Properties** blade and finding the **Infrastructure** resource group.  This resource group contains the virtual network that needs to be paired with the Azure Managed Lustre virtual network. It matches the pattern **MC\_&lt;aks-rg-name&rt;\_&lt;aks-cluster-name&rt;\_&lt;region&rt;**.
```
Comment (Jak-MS) [inline azure-managed-lustre/use-csi-driver-kubernetes.md:113]:
```suggestion
The AKS virtual network is created in a separate resource group from the AKS cluster's resource group. You can find the name of this resource group by going to your AKS cluster in the Azure Portal choosing the **Properties** blade and finding the **Infrastructure** resource group.  This resource group contains the virtual network that needs to be paired with the Azure Managed Lustre virtual network. It matches the pattern **MC\_\<aks-rg-name\>\_\<aks-cluster-name\>\_\<region\>**.
```
The right side angle brackets weren't displaying correctly so I've made changes to fix them... but @jeffbearer - please check that this is the correct pattern: 

MC\_\<aks-rg-name\>\_\<aks-cluster-name\>\_\<region\>

**Comment by Jak-MS** [comment]
@jeffbearer @pauljewellmsft - i've made a few syntax corrections so we can merge the PR. But there is one thing I need you to check ... (which has no been committed yet). see above. 

tag me and sign off when this is ready for merge.

#hold-off

**Comment by jeffbearer** [comment]
@Jak-MS incorporated the last changes and pushed.  let me know if there is anything else needed before merge.

**Pull request merged**